src/ft/mod.rs

```diff
@@ -409,9 +415,32 @@ impl FreeTypeRasterizer {
                     }
 
                     let pattern = font_pattern.clone();
-                    let key = self.face_from_pattern(&pattern, font_key)?.unwrap();
+                    if let Some(ft_face_location) = pattern.ft_face_location(0) {
+                        if self.faces.get(&font_key).is_some() {
+                            return Ok(font_key);
+                        }
+
+                        trace!(
+                            "Got font path={:?}, index={:?}",
+                            ft_face_location.path,
+                            ft_face_location.index
+                        );
+
+                        let ft_face = match self.ft_faces.get(&ft_face_location) {
+                            Some(ft_face) => Rc::clone(ft_face),
+                            None => self.load_ft_face(ft_face_location)?,
+                        };
 
-                    return Ok(key);
+                        let face = FaceLoadingProperties::new(&pattern, ft_face);
+
+                        debug!("Loaded Face {:?}", face);
+
+                        self.faces.insert(font_key, face);
+
+                        return Ok(font_key);
+                    } else {
+                        continue;
+                    }
```

Comment:
There is no reason for this additional code duplication. This is not a proper solution to work around the borrowchecking error